test/e2e_node/image_conformance_test.go

```diff
@@ -84,55 +86,45 @@ var _ = Describe("Image Container Conformance Test", func() {
 		})
 		Context("when testing image that does not exist", func() {
 			var conformImages []ConformanceImage
-			BeforeEach(func() {
-				invalidImageTags := []string{
-					// nonexistent image registry
-					"foo.com/foo/fooimage",
-					// nonexistent image
-					"gcr.io/google_containers/not_exist",
-					// TODO(random-liu): Add test for image pulling credential
-				}
+			invalidImageTags := []string{
+				// nonexistent image registry
+				"foo.com/foo/fooimage",
+				// nonexistent image
+				"gcr.io/google_containers/not_exist",
+				// TODO(random-liu): Add test for image pulling credential
+			}
+			It("should ignore pull failures", func() {
 				for _, invalidImageTag := range invalidImageTags {
 					conformImage, _ := NewConformanceImage("docker", invalidImageTag)
 					// Pulling images from gcr.io is flaky, so retry to make sure failure is not caused by flaky.
-					Consistently(func() error {
-						return conformImage.Pull()
-					}, imageRetryTimeout, imagePullInterval).Should(HaveOccurred())
+					Expect(conformImage.Pull()).Should(HaveOccurred())
 					conformImages = append(conformImages, conformImage)
 				}
-			})
 
-			It("it should not present images [Conformance]", func() {
-				for _, conformImage := range conformImages {
-					present, err := conformImage.Present()
+				By("not presenting images [Conformance]", func() {
+					for _, conformImage := range conformImages {
+						present, err := conformImage.Present()
+						Expect(err).ShouldNot(HaveOccurred())
```

Comment:
Offline discussed with @pwittrock, the test only tests images not exist, so there is no cleanup issue. Even though the following `By` clauses are skipped, that's fine.